pkgs/games/lix/default.nix

```diff
@@ -0,0 +1,144 @@
+    }
+    )
+    (fetchurl {
+      url = "https://github.com/s-ludwig/taggedalgebraic/archive/v${_d_taggedalgebraic_ver}.tar.gz";
+      sha256 = "sha256-KtAI2JCTSZLwkAjfFg2OIcFaG59GfLm3Xh5wGgU7TjE=";
+    }
+    )
+    (fetchurl {
+      url = "https://github.com/atilaneves/unit-threaded/archive/v${_d_unit_threaded_ver}.tar.gz";
+      sha256 = "sha256-uG1EAk7ky/fYGIG4emKIt5TJwkTd0i7OJ4agqf8/nkI=";
+    }
+    )
+  ];
+
+  sourceRoot = ".";
+  nativeBuildInputs = [ dub ldc makeWrapper ];
+  buildInputs = [ pkg-config allegro5 enet ];
+
+  prePatch = ''
+    # we need to set the directory for the data in this source variable
+    substituteInPlace ./LixD-0.10.8/src/file/filename/fhs.d \
+      --replace "enum customReadOnlyDir = \"\";" "enum customReadOnlyDir = \"$out/share\";"
+
+    substituteInPlace ./LixD-0.10.8/data/desktop/com.lixgame.Lix.desktop \
+     --replace "Icon=com.lixgame.Lix" "Icon=lix"
+  '';
+
+  configurePhase = ''
+    runHook preConfigure
+    mkdir home
+    HOME="home" dub add-local SDLang-D-${_d_SDLang_ver} ${_d_SDLang_ver}
+    HOME="home" dub add-local DerelictENet-${_d_derelict_enet_ver} ${_d_derelict_enet_ver}
+    HOME="home" dub add-local enumap-${_d_enumap_ver} ${_d_enumap_ver}
+    HOME="home" dub add-local DAllegro5-4.0.4-5.2.0 ${_d_dallegro_ver}
+    HOME="home" dub add-local optional-${_d_optional_ver} ${_d_optional_ver}
+    HOME="home" dub add-local taggedalgebraic-${_d_taggedalgebraic_ver} ${_d_taggedalgebraic_ver}
+    HOME="home" dub add-local unit-threaded-${_d_unit_threaded_ver} ${_d_unit_threaded_ver}
+    HOME="home" dub add-local DerelictUtil-${_d_derelict_util_ver} ${_d_derelict_util_ver}
+    HOME="home" dub add-local bolts-${_d_bolts_ver} ${_d_bolts_ver}
+    runHook postConfigure
+  '';
+
+
+  buildPhase = ''
+    runHook preBuild
+    cd LixD-${version}
+    HOME="../home" dub --skip-registry=all build -b releaseXDG
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share}
+    mkdir -p $out/share/lix
+    mkdir -p $out/share/{licenses,man,icons,applications}
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    cp -r data $out/share/lix
+    cp -r images $out/share/lix
+    cp -r levels $out/share/lix
+    cp -r ${music} $out/share/lix/music
+    cp bin/lix $out/bin
+
+    install -Dm644 "data/desktop/com.lixgame.Lix.desktop" "$out/share/applications/${pname}.desktop"
+    install -Dm644 "data/images/${pname}_logo.svg" "$out/share/icons/hicolor/scalable/apps/${pname}.svg"
+    install -Dm644 "doc/copying.txt" "$out/share/licenses/${pname}/COPYING"
+    install -Dm644 "doc/${pname}.6" "$out/share/man/man6/${pname}.6"
+
+    # The program loads enet at runtime so we need to set LD_LIBRARY_PATH
+    wrapProgram $out/bin/${pname} --set LD_LIBRARY_PATH "${lib.makeLibraryPath [ enet ]}"
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    ${removeReferencesTo}/bin/remove-references-to -t ${ldc} $out/bin/${pname}
```

Comment:
They may advise you to add `removeReferencesTo` to `nativeBuildInputs` and call the binary directly,but I'm not sure.

Reply:
Yes, please do that.